RollbarNotifier/Sources/RollbarNotifier/DTOs/RollbarDeveloperOptions.m

```diff
@@ -37,6 +39,7 @@ - (instancetype)initWithEnabled:(BOOL)enabled
     self = [super initWithDictionary:@{
         DFK_ENABLED:[NSNumber numberWithBool:enabled],
         DFK_TRANSMIT:[NSNumber numberWithBool:transmit],
+        DFK_SUPPRESS_SDK_INFO_LOGGING:[NSNumber numberWithBool:DEFAULT_SUPPRESS_SDK_INFO_LOGGING_FLAG],
```

Comment:
Why do you inject this option here?
Do you want to force initialization with default setting for the option? If so, how you want to be sure a default value won't be changed in a future and will not resulted in improper functionality?

I would rather drop this line or follow the already existed pattern of overriding. If the latter, this probably would end up to something like:

```objective-c
- (instancetype)initWithEnabled:(BOOL)enabled
                       transmit:(BOOL)transmit
                     logPayload:(BOOL)logPayload
                 payloadLogFile:(NSString *)payloadLogFile
         suppressSdkInfoLogging:(BOOL)suppressSdkInfoLogging {
    
    self = [super initWithDictionary:@{
        DFK_ENABLED:[NSNumber numberWithBool:enabled],
        DFK_TRANSMIT:[NSNumber numberWithBool:transmit],
        DFK_LOG_PAYLOAD:[NSNumber numberWithBool:logPayload],
        DFK_LOG_PAYLOAD_FILE:payloadLogFile,
        DFK_SUPPRESS_SDK_INFO_LOGGING:[NSNumber numberWithBool:suppressSdkInfoLogging]
    }];
    return self;
}

- (instancetype)initWithEnabled:(BOOL)enabled
                       transmit:(BOOL)transmit
                     logPayload:(BOOL)logPayload
                 payloadLogFile:(NSString *)payloadLogFile {
    
    return [self initWithEnabled:enabled
                        transmit:transmit
                      logPayload:logPayload
                  payloadLogFile:payloadLogFile
          suppressSdkInfoLogging:DEFAULT_SUPPRESS_SDK_INFO_LOGGING];
}
```

Reply:
@bxsx , this is the designated initializer implementation and the pattern we established by now regarding initializing config settings objects is: only provide initialization parameters for the most-likely-settings-to-change, the least likely settings to change can be manipulated via config project properties. The reason is that some of the config objects can grow big in terms of modifiable properties we do not want all of them seeded via the initializers - just to keep the designated initializers as compact as possible.